From: Lennart Poettering <lennart@poettering.net>
Date: Thu, 22 May 2014 15:19:46 +0900
Subject: nspawn: restore journal directory is empty check

This undoes part of commit e6a4a517befe559adf6d1dbbadf425c3538849c9.

Instead of removing the error message about non-empty journal bind mount
directories, simply downgrade the message to a warning and proceed.

(cherry picked from commit cdb2b9d05a2f3d649f47bd2ba24eb3fe30b52e92)
(cherry picked from commit 2b1f027f8e194fdb52bc75d3607b01232bdbee15)
---
 src/nspawn/nspawn.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/nspawn/nspawn.c b/src/nspawn/nspawn.c
index d6450010..28e50fba 100644
--- a/src/nspawn/nspawn.c
+++ b/src/nspawn/nspawn.c
@@ -939,6 +939,9 @@ static int setup_journal(const char *directory) {
         } else if (access(p, F_OK) < 0)
                 return 0;
 
+        if (dir_is_empty(q) == 0)
+                log_warning("%s is not empty, proceeding anyway.", q);
+
         r = mkdir_p(q, 0755);
         if (r < 0) {
                 log_error("Failed to create %s: %m", q);
